redash/query_runner/elasticsearch.py

```diff
@@ -317,7 +318,7 @@ def run_query(self, query, user):
             error = None
 
             logger.debug(query)
-            query_params = json.loads(query)
+            query_params = yaml.load(query)
```

Comment:
Good catch that `.safe_load` :100:

Regarding the protection of the parsing code, I was considering this approach:

 - Extend the interface of BaseQueryRunner so it has a method `.parse_query(self, query)`
 - Default implementation checks the runner's `.syntax` attribute to know what to do (json and yaml get parsed, otherwise it's a noop)
 - If an error occurs raise the custom exception